app/components/swiper.js

```diff
@@ -0,0 +1,298 @@
+                >
+                    {pages}
+                </ScrollView>
+            );
+        }
+        return (
+            <ViewPagerAndroid
+                ref={this.refScrollView}
+                {...this.props}
+                initialPage={this.props.initialPage}
+                onScrollBeginDrag={this.onScrollBegin}
+                onPageSelected={this.onScrollEnd}
+                key={pages.length}
+                style={[styles.wrapperAndroid, this.props.style]}
+            >
+                {pages}
+            </ViewPagerAndroid>
+        );
+    };
+
+    renderPagination = () => {
+        if (this.state.total <= 1 || !this.props.showsPagination) {
+            return null;
+        }
+
+        const dots = [];
+        const activeDot = (
+            <View
+                style={[
+                    styles.dotStyle,
+                    {backgroundColor: this.props.activeDotColor || '#007aff'}
+                ]}
+            />
+        );
+        const dot = (
+            <View
+                style={[
+                    styles.dotStyle,
+                    {backgroundColor: this.props.dotColor || 'rgba(0,0,0,.2)'}
+                ]}
+            />
+        );
+        for (let i = 0; i < this.state.total; i++) {
+            if (i === this.state.index) {
+                dots.push(React.cloneElement(activeDot, {key: i}));
+            } else {
+                dots.push(React.cloneElement(dot, {key: i}));
+            }
+        }
+
+        return (
+            <View
+                pointerEvents='none'
+                style={[styles.pagination, this.props.paginationStyle]}
+            >
+                {dots}
+            </View>
+        );
+    };
+
+    scrollToIndex = (index, animated) => {
+        if (this.internals.isScrolling || this.state.total < 2) {
+            return;
+        }
+
+        if (Platform.OS === 'ios') {
+            this.scrollView.scrollTo({x: (index * this.props.width), animated});
+        } else {
+            this.scrollView[animated ? 'setPage' : 'setPageWithoutAnimation'](index);
+        }
+
+        // trigger onScrollEnd manually in android or if not animated
+        if (!animated || Platform.OS === 'android') {
+            setImmediate(() => {
+                this.onScrollEnd({
+                    nativeEvent: {
+                        position: index
+                    }
+                });
+            });
+        }
+    };
+
+    updateIndex = (offset) => {
+        let index = this.state.index;
+        const diff = offset - this.internals.offset;
+        if (!diff) {
+            return;
+        }
+
+        index = parseInt(index + Math.round(diff / this.props.width), 10);
+        this.internals.offset = offset;
+        this.setState({index});
+    };
+
+    render() {
+        const {
+            children,
+            width
+        } = this.props;
+
+        let pages = [];
+        if (this.state.total > 1) {
```

Comment:
I think you can still use React.Children.map with a single child, so this if statement might not be necessary